test/e2e/network/service.go

```diff
@@ -287,7 +287,12 @@ func StartServeHostnameService(ctx context.Context, c clientset.Interface, svc *
 		CreatedPods:          &createdPods,
 		MaxContainerFailures: &maxContainerFailures,
 	}
-	err = e2erc.RunRC(ctx, config)
+
+	dpConfig := testutils.DeploymentConfig{
+		RCConfig: config,
+	}
+
+	err = testutils.RunDeployment(ctx, dpConfig)
```

Comment:
It would be good to get some sig-testing input here or in the corresponding issue. I'd never noticed `testutils` before, and I'm not sure whether it would be better to use stuff from `e2edeployment` instead...

Reply:
autoscaling is not the best place to look at XD

Reply:
filed https://github.com/kubernetes/kubernetes/issues/119774 we will see what sig-testing sais on that.

Reply:
Cc @pholly

Reply:
Deferring to @pohly since he worked a lot of refactoring the e2e tests recently.

Reply:
See https://github.com/kubernetes/kubernetes/issues/119774#issuecomment-1666942854

Reply:
I wrote there:
> Therefore I don't see a need for RunDeployment and would like to see it removed.

So *adding* more usage of it as in this PR goes in the wrong direction.

/lgtm cancel